### PR TITLE
Feature 68 distinguish mode cts files from optional grid_stat cts files

### PR DESCRIPTION
**Pull request by venitahagerty**
## Pull Request Testing ##

This Pull Request is related to METviewer issue 362.

- [ ] Describe testing already performed for these changes:</br>

Ran data sent by John successfully.
Ran METviewer test data successfully.

- [ ] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

Redo the METdatadb/METviewer integration testing from METviewer issue 362.

- [ ] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes or No]**

No documentation changes needed.

- [ ] Do these changes include sufficient testing updates? **[Yes or No]**

- [ ] Will this PR result in changes to the test suite? **[Yes or No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

Does the standard METviewer test data contain all of the new record types added recently?

- [ ] Please complete this pull request review soon.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the PR definition above.
- [ ] Ensure the PR title matches the feature or bugfix branch name.
- [ ] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Organization** level software support **Project** or **Repository** level development cycle **Project**
Select: **Milestone** as the version that will include these changes
- [ ] After submitting the PR, select **Linked issues** with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

**Review by TatianaBurek — APPROVED** [review]
I was able to successfully load data with both grid_stat cts.txt and mode cts.txt.

File: METdbLoad/ush/read_data_files.py [hunk 1]
```diff
@@ -1141,7 +1141,8 @@ def get_lookup(filename):
             lu_type = CN.STAT
         elif lc_filename.endswith(".vsdb"):
             lu_type = CN.VSDB_POINT_STAT
-        elif lc_filename.endswith("cts.txt"):
+        elif (lc_filename.endswith("cts.txt") and 
+              "/mode" in lc_filename):
```
Comment (JohnHalleyGotway) [inline METdbLoad/ush/read_data_files.py]: Thanks @TatianaBurek for testing that this changes works. I do wonder if there's a way to make this check even more explicit? What if the string "/mode" exists somewhere further up in the path of a file named "grid_stat*cts.txt"? That'd break this logic.

Is it possible to do something like:
```
        elif (lc_filename.endswith("cts.txt") and 
              lc_filename.basename().beginswith("mode"):
```
But I don't know the correct syntax for "basename" and "beginswith".

Comment (JohnHalleyGotway) [reply METdbLoad/ush/read_data_files.py]: And I actually just thought of an example of this. In the context of METplus, some use cases are organized into a "model_applications" sub-directory. I don't have any data on hand to test and demonstrate but it does seem that a "model_applications" sub-directory would likely match this!
```
"/mode" in lc_filename
```

Comment (venitahagerty) [reply METdbLoad/ush/read_data_files.py]: I'll use os.path.basename(lc_filename) with beginswith() so we can avoid that case.

**Comment by venitahagerty** [comment]
Made test more specific. Ran again successfully with John's data.

**Pull request merged**